catroid/src/org/catrobat/catroid/formulaeditor/Formula.java

```diff
@@ -272,8 +272,6 @@ public String getResultForComputeDialog(Context context) {
 			} catch (InterpretationException interpretationException) {
 				return "ERROR";
 			}
-			interpretationResult *= 100;
-			interpretationResult = (double) (Math.round(interpretationResult) / 100f);
```

Comment:
You're right and this fixes the problem, of course.
Back then we limited the number of decimals to two to prevent a weird floating point error output (e.g. `0.1+0.1+0.1=0.30000000000000004`).